patch_recurrence/steps.py

```diff
@@ -39,10 +40,53 @@ def generate_patches(scaled_imgs, constants):
         patches.append(img_patches)
     return patches
 
+#..............................................................................................#
+# purva's functions
+
+# exactly like generate_patches without the std deviation condition
+def generate_all_patches(scaled_imgs, constants):
+    patch_size = constants.PATCH_SIZE
+    patches = []
+    for k, sc in enumerate(scaled_imgs):
+        img_patches = []
+        for i in range(sc.shape[0] - patch_size):
+            for j in range(sc.shape[1] - patch_size):
+                raw_patch = sc[i:i + patch_size, j:j + patch_size, :]
+                patch = Patch(
+                    raw_patch=raw_patch,
+                    patch_size=patch_size,
+                )
+                patch.store(sc, [i, j])
+                img_patches.append(patch)
+        patches.append(img_patches)
+    return patches
+
+
+# assigning bucket numbers to each patch
```

Comment:
**nit** - prefer to use docstrings rather than one line `#` comments for function descriptions